From 77079777d257bad0f53b16b004407002e482771d Mon Sep 17 00:00:00 2001
From: Gabriel Fernandez <gabriel.fernandez@foss.st.com>
Date: Tue, 27 Jun 2023 11:55:39 +0200
Subject: [PATCH 0475/1695] clk: stm32-core: remove static for stm32_gate and
 stm32_div ops

Remove static for stm32_gate and stm32_div ops to be use directly
int STM32 RCC drivers.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git d2f5ad4cc28477b02691f5fd5152cf237a890846 clk: stm32-core: remove static for stm32_gate and stm32_div ops]
Change-Id: Ie00bafa257354038479a440c43e2f13694f93fc6
Signed-off-by: Gabriel Fernandez <gabriel.fernandez@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/326475
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Domain-Review: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
Reviewed-by: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
---
 core/drivers/clk/clk-stm32-core.c | 14 +++++++-------
 core/drivers/clk/clk-stm32-core.h |  8 ++++++++
 2 files changed, 15 insertions(+), 7 deletions(-)

diff --git a/core/drivers/clk/clk-stm32-core.c b/core/drivers/clk/clk-stm32-core.c
index cae769443..7258ce7c0 100644
--- a/core/drivers/clk/clk-stm32-core.c
+++ b/core/drivers/clk/clk-stm32-core.c
@@ -62,7 +62,7 @@ TEE_Result stm32_mux_set_parent(uint16_t mux_id, uint8_t sel)
 }
 
 /* STM32 GATE API */
-static void stm32_gate_endisable(uint16_t gate_id, bool enable)
+void stm32_gate_endisable(uint16_t gate_id, bool enable)
 {
 	struct clk_stm32_priv *priv = clk_stm32_get_priv();
 	const struct gate_cfg *gate = &priv->gates[gate_id];
@@ -295,7 +295,7 @@ TEE_Result stm32_div_set_value(uint32_t div_id, uint32_t value)
 	return stm32_gate_wait_ready((uint16_t)divider->ready, true);
 }
 
-static unsigned long stm32_div_get_rate(int div_id, unsigned long prate)
+unsigned long stm32_div_get_rate(int div_id, unsigned long prate)
 {
 	struct clk_stm32_priv *priv = clk_stm32_get_priv();
 	const struct div_cfg *divider = &priv->div[div_id];
@@ -346,7 +346,7 @@ const struct clk_ops clk_stm32_mux_ops = {
 };
 
 /* STM32 GATE clock operators */
-static TEE_Result clk_stm32_gate_enable(struct clk *clk)
+TEE_Result clk_stm32_gate_enable(struct clk *clk)
 {
 	struct clk_stm32_gate_cfg *cfg = clk->priv;
 
@@ -355,14 +355,14 @@ static TEE_Result clk_stm32_gate_enable(struct clk *clk)
 	return TEE_SUCCESS;
 }
 
-static void clk_stm32_gate_disable(struct clk *clk)
+void clk_stm32_gate_disable(struct clk *clk)
 {
 	struct clk_stm32_gate_cfg *cfg = clk->priv;
 
 	stm32_gate_disable(cfg->gate_id);
 }
 
-static bool clk_stm32_gate_is_enabled(struct clk *clk)
+bool clk_stm32_gate_is_enabled(struct clk *clk)
 {
 	struct clk_stm32_gate_cfg *cfg = clk->priv;
 
@@ -375,14 +375,14 @@ const struct clk_ops clk_stm32_gate_ops = {
 	.is_enabled	= clk_stm32_gate_is_enabled,
 };
 
-static TEE_Result clk_stm32_gate_ready_enable(struct clk *clk)
+TEE_Result clk_stm32_gate_ready_enable(struct clk *clk)
 {
 	struct clk_stm32_gate_cfg *cfg = clk->priv;
 
 	return stm32_gate_rdy_enable(cfg->gate_id);
 }
 
-static void clk_stm32_gate_ready_disable(struct clk *clk)
+void clk_stm32_gate_ready_disable(struct clk *clk)
 {
 	struct clk_stm32_gate_cfg *cfg = clk->priv;
 
diff --git a/core/drivers/clk/clk-stm32-core.h b/core/drivers/clk/clk-stm32-core.h
index 9eb97339e..8eaff9797 100644
--- a/core/drivers/clk/clk-stm32-core.h
+++ b/core/drivers/clk/clk-stm32-core.h
@@ -113,6 +113,7 @@ struct clk_stm32_gate_ready_cfg {
 #define NO_DIV		INT32_MAX
 #define NO_GATE		INT32_MAX
 
+void stm32_gate_endisable(uint16_t gate_id, bool enable);
 void stm32_gate_enable(uint16_t gate_id);
 void stm32_gate_disable(uint16_t gate_id);
 bool stm32_gate_is_enabled(uint16_t gate_id);
@@ -123,6 +124,7 @@ TEE_Result stm32_gate_rdy_disable(uint16_t gate_id);
 size_t stm32_mux_get_parent(uint32_t mux_id);
 TEE_Result stm32_mux_set_parent(uint16_t pid, uint8_t sel);
 
+unsigned long stm32_div_get_rate(int div_id, unsigned long prate);
 TEE_Result stm32_div_set_rate(int div_id, unsigned long rate,
 			      unsigned long prate);
 
@@ -132,6 +134,12 @@ TEE_Result stm32_div_set_value(uint32_t div_id, uint32_t value);
 int clk_stm32_parse_fdt_by_name(const void *fdt, int node, const char *name,
 				uint32_t *tab, uint32_t *nb);
 
+TEE_Result clk_stm32_gate_ready_enable(struct clk *clk);
+void clk_stm32_gate_ready_disable(struct clk *clk);
+TEE_Result clk_stm32_gate_enable(struct clk *clk);
+void clk_stm32_gate_disable(struct clk *clk);
+bool clk_stm32_gate_is_enabled(struct clk *clk);
+
 unsigned long clk_stm32_divider_get_rate(struct clk *clk,
 					 unsigned long parent_rate);
 
-- 
2.47.2

